src/cli/one_helper/oneprovider_helper.rb

```diff
@@ -14,203 +14,418 @@
 
-            raise 'Name not found' unless template['name']
+            tmp.write(JSON.pretty_generate(body))
+            tmp.flush
 
-            OneProvision::Terraform.p_load
+            if ENV['EDITOR']
+                editor_path = ENV['EDITOR']
+            else
+                editor_path = OpenNebulaHelper::EDITOR_PATH
+            end
 
-            unless OneProvision::Terraform.exist?(template['provider'])
-                raise 'Invalid provider, available providers: ' \
-                      "#{OneProvision::Terraform.providers.join(', ')}"
+            system("#{editor_path} #{path}")
+
+            unless $CHILD_STATUS.exitstatus.zero?
+                STDERR.puts 'Editor not defined'
+                exit(-1)
             end
 
-            return template if template['provider'] == 'dummy'
+            tmp.close
+        end
 
-            raise 'Connection info not found' unless template['connection']
+        body = read_json_input(path)
+        response = client.update_provider(provider_id, body)
 
-            template
-        rescue StandardError => e
-            OpenNebula::Error.new("ERROR: #{e}")
+        if CloudClient.is_error?(response)
+            [response[:err_code], response[:message]]
+        else
+            0
         end
     end
 
-    # Returns a new provider object
-    #
-    # @param id [Integer] Provider ID
-    def factory(id = nil)
-        if id
-            OneProvision::Provider.new_with_id(id, @client)
+    def read_json_input(file)
+        if file
+            begin
+                content = File.read(file)
+            rescue Errno::ENOENT
+                STDERR.puts "File not found: #{file}"
+                exit(-1)
+            end
         else
-            OneProvision::Provider.new(OneProvision::Provider.build_xml,
-                                       @client)
+            stdin = OpenNebulaHelper.read_stdin
+            return if stdin.empty?
+
+            content = stdin
         end
+
+        begin
+            JSON.parse(content, :symbolize_names => true)
+        rescue JSON::ParserError => e
+            source = file ? "file: #{file}" : 'stdin'
+            STDERR.puts "Invalid JSON in #{source} - #{e.message}"
+            exit(-1)
+        end
+    end
+
+    def get_user_values(user_inputs)
+        return {} if user_inputs.nil? || user_inputs.empty?
+
+        ask_user_inputs(user_inputs)
     end
 
-    # Returns provider pool
-    def factory_pool(_)
-        OneProvision::ProviderPool.new(@client)
+    def ask_user_inputs(inputs)
+        puts 'There are some parameters that require user input.'
+
+        answers = {}
+
+        inputs.each do |input|
+            name        = input[:name]
+            description = input[:description] || ''
+            type        = input[:type]
+            default     = input[:default]
+            match       = input[:match]
+
+            puts "  * (#{name}) #{description} [type: #{type}]"
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

A cross-site scripting (XSS) vulnerability was detected. User-controlled data is being output without proper sanitization, allowing an attacker to inject malicious scripts. This could lead to session hijacking, malware installation, or phishing attacks. Always sanitize and encode user inputs before including them in the output, using context-appropriate encoding methods. Learn More - https://cwe.mitre.org/data/definitions/79.html.